openapi2beans/pkg/embedded/embedded.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package embedded
+
+import (
+	"embed"
+
+	"github.com/cbroglie/mustache"
+	openapi2beans_errors "github.com/dev-galasa/buildutils/openapi2beans/pkg/errors"
+)
+
+// Embed all the template files into the go executable, so there are no extra files
+// we need to ship/install/locate on the target machine.
+// We can access the "embedded" file system as if they are normal files.
+//
+//go:embed templates/*
+var embeddedFileSystem embed.FS
+
+// An instance of the ReadOnlyFileSystem interface, set once, used many times.
+// It just delegates to teh embed.FS
+var readOnlyFileSystem ReadOnlyFileSystem
+
+type templates struct {
+	JavaClassTemplate         *mustache.Template
+	JavaEnumTemplate          *mustache.Template
+}
+
+const (
+	JAVA_CLASS_TEMPLATE_FILEPATH = "templates/JavaClassTemplate.mustache"
+	JAVA_ENUM_TEMPLATE_FILEPATH  = "templates/JavaEnumTemplate.mustache"
+)
+
+var (
+	templatesCache *templates = nil
+)
+
+func GetReadOnlyFileSystem() ReadOnlyFileSystem {
+	if readOnlyFileSystem == nil {
+		readOnlyFileSystem = NewReadOnlyFileSystem()
+	}
+	return readOnlyFileSystem
+}
+
+func GetJavaClassTemplate() (*mustache.Template, error) {
+	var err error
+	fs := GetReadOnlyFileSystem()
+	// Note: The cache is set when we read the versions from the embedded file.
+	templatesCache, err = readTemplatesFromEmbeddedFiles(fs, templatesCache)
+	var template *mustache.Template
+	if err == nil {
+		template = templatesCache.JavaClassTemplate
+	} else {
+		err = openapi2beans_errors.NewError("Failed to read templates from embedded file. Reason is: %s", err.Error())
+	}
+	return template, err
+}
+
+func GetJavaEnumTemplate() (*mustache.Template, error) {
```

Comment:
This is the same code as above ? Perhaps they should both call a 2nd function which takes a parameter or two ?